src/eval.c

```diff
@@ -5394,6 +5465,774 @@ get_lit_string_tv(char_u **arg, typval_T *rettv, int evaluate)
+    lit_string[0] = '\'';
+    memcpy(lit_string + 1, content, content_length);
+    lit_string[whole_length - 1] = '\'';
+
+    return lit_string;
+}
+
+/*
+ * enclose_by_double_quote("a content") ==# "\"a content \""
+ *
+ * This result MUST BE `free`d later!
+ */
+    static char_u*
+enclose_by_double_quote(char_u* content)
+{
+    size_t	content_length = STRLEN(content);
+    size_t	whole_length = content_length + 2;
+
+    char_u* lit_string = alloc_terminated(whole_length);
+    lit_string[0] = '"';
+    memcpy(lit_string + 1, content, content_length);
+    lit_string[whole_length - 1] = '"';
+
+    return lit_string;
+}
+
+    static int
+is_started_with_quote(char_u* x)
+{
+    return
+	x[0] == '\'' ||
+	x[0] == '"';
+}
+
+/*
+ * This result MUST BE `free`d later!
+ */
+    static char_u*
+disclose_quote(char_u* quoted)
+{
+    const size_t	QUOTES_LENGTH = 2;  // strlen("''")
+    size_t		result_length = STRLEN(quoted) - QUOTES_LENGTH;
+    char_u*		result;
+
+    if ((quoted[0] != '\'') && (quoted[0] != '"'))
+    {
+	return NULL;
+    }
+    result = alloc_terminated(result_length);
+
+    memcpy(result, quoted + 1, result_length);
+    return result;
+}
+
+/*
+ * This result MUST BE `free`d later!
+ */
+    static char_u*
+duplicate_single_quote(char_u x)
+{
+    char_u*	xs;
+
+    if (x == '\'')
+    {
+	return vim_strsave((char_u*) "''");
+    }
+
+    xs = alloc_terminated(1);
+    xs[0] = x;
+    return xs;
+}
+
+/*
+ * This result MUST BE `free`d later!
+ */
+    static char_u*
+dont_fix(char_u x)
+{
+    char_u*	xs = alloc_terminated(1);
+    xs[0] = x;
+    return xs;
+}
+
+    static int
+is_closing_single_quote(char_u* x)
+{
+    const char_u	DELIMITTERS[] = {NUL, ',', ')', ' '};
+    const size_t	DELIMITTERS_SIZE = 4;
+    size_t		i;
+
+    if (x[0] != '\'')
+    {
+	return FALSE;
+    }
+
+    for (i = 0; i < DELIMITTERS_SIZE; ++i)
+    {
+	if (x[1] == DELIMITTERS[i])
+	{
+	    return TRUE;
+	}
+    }
+
+    return FALSE;
+}
+
+/*
+ * NOTICE: x[-1] must be readable.
+ */
+    static int
+is_closing_double_quote(char_u* x)
+{
+    return (x[0] == '"') && (x[-1] != '\\');
+}
+
+/*
+ * Allocate a variable for a $'string ${var} $x' constant.
+ * Return OK or FAIL.
+ *
+ * enclose() encloses a content by '' (or "").
+ * fix_quote() fixes quotes ' to ''. this fixes a problem that string() stringifies {"x": 10} to {'x': 10}, ["a", "b"] to ['a', 'b'].
+ * is_closing_quote() checks a char is ' (or ").
+ */
+    static int
+get_template_string_tv(
+	char_u* (*enclose)(char_u*),
+	char_u* (*disclose)(char_u*),
+	char_u* (*fix_quote)(char_u),
+	int (*is_closing_quote)(char_u*),
+	char_u **arg,
+	typval_T *rettv,
+	int evaluate)
+{
+    char_u*	string;
+    char_u*	origin_string;
+    char_u*	template;
+
+    // NOTE: At here, "${}" of "${" "expr" "}", that called 'embedding', and "expr" called 'embedded'.
+
+    template = read_template_string(is_closing_quote, *arg);
+    if (template == NULL) {
```

Comment:
This `{` should be the next line. (or should be removed, because this block is only one line.)

Reply:
:sunglasses:  :+1:
I did `/\v\) \{/` and `f s<CR>` all!